bin/node/test-runner-example/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+            pallet_transaction_payment::ChargeTransactionPayment::<Self::Runtime>::from(0),
+        )
+    }
+
+    fn create_client_parts(
+        config: &Configuration,
+    ) -> Result<
+        (
+            Arc<TFullClient<Self::Block, Self::RuntimeApi, Self::Executor>>,
+            Arc<TFullBackend<Self::Block>>,
+            SyncCryptoStorePtr,
+            TaskManager,
+            InherentDataProviders,
+            Option<
+                Box<
+                    dyn ConsensusDataProvider<
+                        Self::Block,
+                        Transaction = sp_api::TransactionFor<
+                            TFullClient<Self::Block, Self::RuntimeApi, Self::Executor>,
+                            Self::Block,
+                        >,
+                    >,
+                >,
+            >,
+            Self::SelectChain,
+            Self::BlockImport,
+        ),
+        sc_service::Error,
+    > {
+        let (client, backend, keystore, task_manager) =
+            new_full_parts::<Self::Block, Self::RuntimeApi, Self::Executor>(config, None)?;
+        let client = Arc::new(client);
+
+        let inherent_providers = InherentDataProviders::new();
+        let select_chain = sc_consensus::LongestChain::new(backend.clone());
+
+        let (grandpa_block_import, ..) =
+            grandpa::block_import(
+                client.clone(),
+                &(client.clone() as Arc<_>),
+                select_chain.clone(),
+                None
+            )?;
+
+        let (block_import, babe_link) = sc_consensus_babe::block_import(
+            sc_consensus_babe::Config::get_or_compute(&*client)?,
+            grandpa_block_import,
+            client.clone(),
+        )?;
+
+        let consensus_data_provider = BabeConsensusDataProvider::new(
+            client.clone(),
+            keystore.sync_keystore(),
+            &inherent_providers,
+            babe_link.epoch_changes().clone(),
+            vec![(AuthorityId::from(Alice.public()), 1000)],
+        )
+            .expect("failed to create ConsensusDataProvider");
+
+        Ok((
+            client,
+            backend,
+            keystore.sync_keystore(),
+            task_manager,
+            inherent_providers,
+            Some(Box::new(consensus_data_provider)),
+            select_chain,
+            block_import,
+        ))
+    }
+
+    fn dispatch_with_root(call: <Self::Runtime as frame_system::Config>::Call, node: &mut Node<Self>) {
+        let alice = MultiSigner::from(Alice.public()).into_account();
+        let call = pallet_sudo::Call::sudo(Box::new(call));
+        node.submit_extrinsic(call, alice);
+        node.seal_blocks(1);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_runner::NodeConfig;
+    use log::LevelFilter;
+    use sc_client_api::execution_extensions::ExecutionStrategies;
+    use node_cli::chain_spec::development_config;
+
+    #[test]
+    #[ignore]
```

Comment:
This is not a good shwocase with an ignored test. Why is that ignored?

Reply:
I can only guess that because it needs a snapshot of a db.

Reply:
It doesn't seem to pass on CI, there's a feature leak somewhere because of the `--features runtime-benchmarks` used in running the tests. The temporary solution is to ignore it while we try to figure out where it's coming from.

Reply:
What do you want to figure out? Where we enable this feature? It is enabled for all tests.

You need to add the required host functions for this. Just do this and your problem will be solved.

Reply:
(hmm okay then this needs to be solved and should not be `#[ignore]`d)

Reply:
fixed